squiggy/models/asset.py

```diff
@@ -507,6 +507,10 @@ def _build_where_clause(filters, include_hidden, params, session):
         where_clause += ' AND a.visible = TRUE'
     if not session.is_admin and not session.is_teaching:
         where_clause += ' AND (a.visible = TRUE OR a.id = ANY(:my_asset_ids))'
+    if session.course.protects_assets_per_section and session.is_student:
+        where_clause += """ AND (to_jsonb(u.canvas_course_sections) ?| (select canvas_course_sections from users where id = :user_id)
+            OR u.canvas_course_role <> 'Student'
+        )"""
```

Comment:
@lyttam - What is the requirement for 'Observer' role?  We've had to account for that role in SuiteC but not sure how it plays into this feature. If Observers have no section-affiliation then that might settle it.

Reply:
I think Observers tend to be more like teachers or TAs and we wouldn't need to silo them. But another role we've seen that probably should be handled the same as 'Student' is 'Learner'.

Reply:
You sure about Observers @pauline2k ?  The prevailing logic in squiggy, as far as Teena knows, is that Observers have restrictions similar to Students in all the tools.

Reply:
The observer role does sometimes get used in cases such as one-instructor-inviting-another-to-review-their-course-materials, but observers don't have edit access (they can't even submit assignments), so in that senses they're more like students.

They're intended to be used in k12 so that a parent can observe a particular student. We don't support or allow that use case (or encourage their use at at all), so I'm in favor of whatever method of handling the observer role is simplest to implement.

Reply:
I think https://github.com/ets-berkeley-edu/squiggy/blob/dev/squiggy/lib/util.py is our best guide.

"Learner" and "Student" are student roles and should be siloed.
"Teacher," "Instructor" and "Admin" have superpowers and aren't siloed.
"Observer" doesn't get teaching or admin superpowers, but given the position people with this role tend to occupy (often instructional designers I think?) I don't think it makes sense to put a student-specific restriction on them.

Reply:
Note also the that those role checks are doing substring comparisons because some roles, like Instructor, actually come from Canvas with a silly IMS schema designation 'urn:lti:role:ims/lis/Instructor'. Checking the squiggy-prod database might be the best guide as to how student and learner roles are currently showing up.